helpdesk_mgmt_stage_server_action/tests/test_helpdesk_ticket_stage.py

```diff
@@ -1,96 +1,45 @@
-                        },
-                    )
-                ],
-            }
-        )
-        cls.helpdesk_ticket_stage_1 = cls.HelpdeskTicketStage.create(
-            {"name": "Stage 1", "sequence": 1}
-        )
-        cls.helpdesk_ticket_stage_2 = cls.HelpdeskTicketStage.create(
-            {
-                "name": "Stage 2",
-                "action_id": cls.server_action_helpdesk_ticket.id,
-                "sequence": 2,
-            }
-        )
-
-    def test_helpdesk_ticket_create(self):
-        self.helpdesk_ticket_1 = self.HelpdeskTicket.create(
+        cls.HelpdeskTicketTag = cls.env["helpdesk.ticket.tag"]
+        cls.ServerAction = cls.env["ir.actions.server"]
+        cls.create_tag_action = cls.env["ir.actions.server"].create(
             {
-                "name": "Helpdesk Ticket 1",
-                "stage_id": self.helpdesk_ticket_stage_2.id,
-                "description": "Helpdesk Ticket Description",
+                "model_id": cls.env["ir.model"]._get_id("helpdesk.ticket.tag"),
+                "crud_model_id": cls.env["ir.model"]._get_id("helpdesk.ticket.tag"),
+                "name": "Create new helpdesk tag",
+                "value": "New helpdesk tag",
+                "state": "object_create",
             }
         )
-        self.assertEqual(self.helpdesk_ticket_1.user_id, self.env.user)
-
-    def test_helpdesk_ticket_write(self):
-        self.helpdesk_ticket_2 = self.HelpdeskTicket.create(
-            {
-                "name": "Helpdesk Ticket 2",
-                "stage_id": self.helpdesk_ticket_stage_1.id,
-                "description": "Helpdesk Ticket Description",
-            }
+        cls.stage_1 = cls.HelpdeskTicketStage.create(
+            {"name": "Stage 1", "sequence": 1, "action_id": cls.create_tag_action.id}
         )
-        self.helpdesk_ticket_3 = self.HelpdeskTicket.create(
+        cls.ticket = cls.HelpdeskTicket.create(
             {
-                "name": "Helpdesk Ticket 3",
-                "stage_id": self.helpdesk_ticket_stage_1.id,
-                "description": "Helpdesk Ticket Description",
+                "name": "Create a tag ticket",
+                "description": "Ticket Description",
             }
         )
-        self.assertNotEqual(self.helpdesk_ticket_2.user_id, self.env.user)
-        self.assertNotEqual(self.helpdesk_ticket_3.user_id, self.env.user)
-        self.helpdesk_ticket_2.write({"stage_id": self.helpdesk_ticket_stage_2.id})
-        self.helpdesk_ticket_3.write({"stage_id": self.helpdesk_ticket_stage_2.id})
-        self.assertEqual(self.helpdesk_ticket_2.user_id, self.env.user)
-        self.assertEqual(self.helpdesk_ticket_3.user_id, self.env.user)
-        self.helpdesk_ticket_3.write({"user_id": False})
-        self.helpdesk_ticket_3.write({"stage_id": self.helpdesk_ticket_stage_2.id})
-        self.assertFalse(self.helpdesk_ticket_3.user_id)
 
-    def test_helpdesk_ticket_without_stage(self):
```

Comment:
I would very much prefer to keep the existing unit tests during a migration. It helps to ensure the behavior is not unwantedly modified.

It's ok to refactor, but the same features should be tested (unless we decide to change them of course, but I think this isn't the case). I think a feature here is lost: not triggering the action when the stage doesn't actually change, as mentioned in the previous comment